var/spack/repos/builtin/packages/libtirpc/package.py

```diff
@@ -21,10 +22,11 @@ class Libtirpc(AutotoolsPackage):
 
     # Remove -pipe flag to compiler in Makefiles when using nvhpc
     patch("libtirpc-remove-pipe-flag-for-nvhpc.patch", when="%nvhpc")
+    patch("macos-1.3.3.patch", when="@1.3.3 platform=darwin")
```

Comment:
Can you add a brief comment above the `patch` directive explaining:
1. What is the effect of the patch
2. Why it is needed
3. Where it was taken from

?